lib/workflow/stores/mongo.js

```diff
@@ -677,11 +695,22 @@ function mongoStoreFactory(waterline, Promise, Constants, Errors, assert, _) {
 
         var query = {
             graphId: data.graphId,
-            reachable: true
+            reachable: true,
+            $or:[]
         };
-        query['dependencies.' + data.taskId] = {
+ 
+        var dependenciesItr = {};
+        dependenciesItr['dependencies.' + data.taskId] = {
             $in: _.difference(Constants.Task.FinishedStates, [data.state])
         };
+        query.$or.push(dependenciesItr);
+
+        delete dependenciesItr['dependencies.' + data.taskId];
+        dependenciesItr['dependencies.anyOf.' + data.taskId] = {
+            $in: _.difference(Constants.Task.FinishedStates, [data.state])
+        };
+        query.$or.push(dependenciesItr);
```

Comment:
Don't share the same variable `dependenciesItr` for different criteria, Below simple test will show you the query is actually not what you want:
```javascript
var or = [];                                                                                                                                              
var depItr = {};                                                                                                                                          
depItr = { a: 1 };                                                                                                                                        
or.push(depItr);                                                                                                                                          
                                                                                                                                                          
delete depItr.a;                                                                                                                                          
depItr = { b: 2 };                                                                                                                                        
or.push(depItr);                                                                                                                                          
                                                                                                                                                          
console.log(or); 

//The output is [ {}, { b: 2 } ] , not [ { a: 1 }, { b: 2 } ] !!!!
```